pkg/handlers/s3_handler.go

```diff
@@ -103,6 +104,13 @@ func (h *s3Handler) OnAdd(ns string, vc kvcv1.VolumeConfig, controllerRef metav1
 		recursiveFlag = "--recursive"
 	}
 
+	s3URL, err := url.Parse(vc.SourceURL)
+	if err != nil {
+		glog.Warningf("[s3-handler] Error while parsing URL [%s], %v", vc.SourceURL, err)
+	}
```

Comment:
I guess we should error out return from here since `s3URL.Host` is going to fail if the err is not nil.